src/core/border_router/routing_manager.cpp

```diff
@@ -3762,6 +3762,12 @@ extern "C" void otPlatBorderRoutingProcessIcmp6Ra(otInstance *aInstance, const u
 {
     AsCoreType(aInstance).Get<BorderRouter::RoutingManager>().ProcessPlatformGeneratedRa(aMessage, aLength);
 }
+
+extern "C" OT_TOOL_WEAK void otPlatBorderRoutingRequestDhcp6Pd(otInstance *aInstance, bool aRequest)
```

Comment:
In this case, some platforms need to implement this API to control it's platform PD, while some devices don't need to implement this API(eg. Nest WiFi Pro). So I think define a WEAK is more feasible here. 
For the incorrect implementations, I think it's out of our control, we've already provided the usage guide in the API description, developers should just follow it, if they misunderstood it they can come and ask us?

Reply:
In this case, some platforms need to implement this API to control it's platform PD, while some devices don't need to implement this API(eg. some CPE). So I think define a WEAK is more feasible here.
For the incorrect implementations, I think it's out of our control, we've already provided the usage guide in the API description, developers should just follow it, if they misunderstood it they can come and ask us?

Reply:
> Providing empty weak implementation of platform API in OT core mask potential errors (e.g., incorrect implementation using wrong name or wrong param by platform layer).

Makes sense.

I'm thinking if it is "optional" or it "can have an empty implementation on some platforms". I can see some CPE devices in the market does not have a DHCPv6 PD server embedded, and if they integrate the PD feature, they will use a static prefix allocation, and they can assume there are no second BR in the network that provides GUA prefixes (so they don't have to withdrawn the prefix, and nowhere to return the lease).

Previously I inteprete the above seenrio as "optional", but now I think maybe a macro like "`OPENTHREAD_BORDER_ROUTING_EMPTY_DHCP6_PD_REQUESTOR`" (or other names) to treat this as a "special case" instead of a "optional API" since the above case is not a standard path?

Reply:
> In this case, some platforms need to implement this API to control it's platform PD, while some devices don't need to implement this API(eg. some CPE). So I think define a WEAK is more feasible here.

I think we can leave this to platform, if the platform has nothing to do, they can provide empty implementation of this API. 

> For the incorrect implementations, I think it's out of our control, we've already provided the usage guide in the API description, developers should just follow it, if they misunderstood it they can come and ask us?

The difference is when we provide a weak implementation, and platform defines improper name/format for the API, the build will succeed and we then need to debug the issue at run-time. But if we don't provide a weak implementation, we get an error at build time which is much easier to fix.

> Previously I inteprete the above seenrio as "optional", but now I think maybe a macro like "OPENTHREAD_BORDER_ROUTING_EMPTY_DHCP6_PD_REQUESTOR" (or other names) to treat this as a "special case" instead of a "optional API" since the above case is not a standard path?

Adding a config for this make sense, but may be better to add it in platform. Since such a config is really  confiuguration the platform behavior and not core OT stack `BorderRouting` behavior. We can consider adding such a config in `posix` platform.

Here are my suggestions on this:
- From OT platform API, we require this API to be provided by platform when  `BORDER_ROUTING_DHCP6_PD_ENABLE` is enabled (dont treat as optional).
- We leave it up to platform to decide whether to provide an empty implementation or do something.
- We can add  the empty implementation of this API in `posix` platform source files 
   - And an add a config in `POSIX` to indicate whether `posix` source code should provide an empty implementation for it and leave it to be defined by `ot-br-posix` module.

Thoughst?